samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -264,110 +275,12 @@ private Windows() { }
       c.add(m);
       return c;
     };
-    return keyedSessionWindow(keyFn, sessionGap, aggregator);
-  }
-
-
-  /**
-   * Creates a {@link Window} that groups incoming messages into a single global window. This window does not have a
-   * default trigger. The triggering behavior must be specified by setting an early trigger.
-   *
-   * <p>The below example computes the maximum value over a count based window. The window emits {@link WindowPane}s when
-   * there are either 50 messages in the window pane or when 10 seconds have passed since the first message in the pane.
-   *
-   * <pre> {@code
-   *    MessageStream<Long> stream = ...;
-   *    BiFunction<Long, Long, Long> maxAggregator = (m, c)-> Math.max(m, c);
-   *    MessageStream<WindowPane<Void, Long>> windowedStream = stream.window(Windows.globalWindow(maxAggregator)
-   *      .setEarlyTriggers(Triggers.repeat(Triggers.any(Triggers.count(50), Triggers.timeSinceFirstMessage(Duration.ofSeconds(10))))))
-   * }
-   * </pre>
-   *
-   * @param foldFn the function to aggregate messages in the {@link WindowPane}
-   * @param <M> the type of message
-   * @param <WV> type of the output value in the {@link WindowPane}
-   * @return the created {@link Window} function.
-   */
-  public static <M, WV> Window<M, Void, WV> globalWindow(BiFunction<M, WV, WV> foldFn) {
-    return new WindowInternal<>(null, foldFn, null, null);
-  }
 
-  /**
-   * Creates a {@link Window} that groups incoming messages into a single global window. This window does not have a
-   * default trigger. The triggering behavior must be specified by setting an early trigger.
-   *
-   * The below example groups the stream into count based windows that trigger every 50 messages or every 10 minutes.
-   * <pre> {@code
-   *    MessageStream<Long> stream = ...;
-   *    MessageStream<WindowPane<Void, Collection<Long>> windowedStream = stream.window(Windows.globalWindow()
-   *      .setEarlyTrigger(Triggers.repeat(Triggers.any(Triggers.count(50), Triggers.timeSinceFirstMessage(Duration.ofSeconds(10))))))
-   * }
-   * </pre>
-   *
-   * @param <M> the type of message
-   * @return the created {@link Window} function.
-   */
-  public static <M> Window<M, Void, Collection<M>> globalWindow() {
-    BiFunction<M, Collection<M>, Collection<M>> aggregator = (m, c) -> {
-      c.add(m);
-      return c;
+    Supplier<Collection<M>> initialValue = () -> {
```

Comment:
I've refactored the lambda to be a one-liner as
`() -> new ArrayList<>()` . 

Let me know if you still prefer the inner function class approach.